app/models/grda_warehouse/ch_enrollment.rb

```diff
@@ -74,5 +74,324 @@ def self.update_existing!
         )
       end
     end
+
+    # Accept an optional date which will be used for extending the homeless
+    # range if the project is a homeless project
+    def self.chronically_homeless_at_start?(enrollment, date: enrollment.EntryDate)
+      chronically_homeless_at_start(enrollment, date: date) == :yes
+    end
+
+    # Was the client chronically homeless at the start of this enrollment?
+    # Optionally accepts a date to use for "CH at a point-in-time" calculation.
+    #
+    # @return [Symbol] :yes, :no, :dk_or_r, or :missing
+    def self.chronically_homeless_at_start(enrollment, date: enrollment.EntryDate)
+      chronically_homeless_at_start_steps(enrollment, date: date).last[:result]
+    end
+
+    # Was the client chronically homeless at the start of this enrollment?
+    # Optionally accepts a date to use for "CH at a point-in-time" calculation.
+    #
+    # @return [Array] all steps evaluated with result and display values.
+    #
+    # Each step in the array has this shape:
+    # {
+    #    result: Symbol, true, false, or nil
+    #    display_value: value to display in chronic-at-entry explanation table
+    #    line: [Number] line number in HUD calculation
+    # }
+    #
+    # Result "nil" means continue processing
+    # Result "false" means continue processing and skip branch
+    # Result "true" means continue processing and enter branch
```

Comment:
This is the only bit that isn't sitting well with me.  It feels like the methods should return something like
`:continue` `:skip` `:enter`, but I'm not sold on those either.  Mostly I don't like the `nil` being used to make a decision when in combination with true and false, since in most situations nil and false behave similarly.  It's really elegant as written, so I'm not sold either way.  Maybe we can discuss tomorrow.

Reply:
Yeah I hear that, don't love this either. It may be confusing to come back to. I can change it so `result` is always one of `:yes, :no, :dk_or_r, :missing, :enter, :continue`, and then instead of using truthy/false checks to decide when to return, we need to check something like `is_terminating?(sym) => sym.in?([:yes, :no, :dk_or_r, :missing])` instead. I think it is clearer.